packages/blade/src/components/Alert/_decisions/decisions.md

```diff
@@ -0,0 +1,162 @@
+
+### Alert.Description
+
+| Prop       | Type          | Default     | Description | Required |
+| ---------- | ------------- | ----------- | ----------- | -------- |
+| `children` | `string`, JSX | `undefined` | Content \*  | ✅       |
+
+### Alert.PrimaryAction
+
+`Alert.PrimaryAction` renders and accepts a subset of `Button` (primary) props:
+
+| Prop       | Type       | Default     | Description                            | Required |
+| ---------- | ---------- | ----------- | -------------------------------------- | -------- |
+| `children` | `string`   | `undefined` | Content, required here unlike `Button` | ✅       |
+| `onClick`  | `function` | `undefined` | Callback                               | ✅       |
+
+### Alert.SecondaryAction
+
+`Alert.SecondaryAction` renders and accepts a subset of `Link` (button) props:
+
+| Prop       | Type          | Default     | Description | Required |
+| ---------- | ------------- | ----------- | ----------- | -------- |
+| `children` | `string`, JSX | `undefined` | Content     | ✅       |
+| `href`     | `string`      | `undefined` | Hyperlink   |          |
+| `onClick`  | `function`    | `undefined` | Callback    |          |
+
+There'll also be an internal `BaseAlert` component that handles rendering web or native version similar to other Blade components. API would be similar to `Alert`.
+
+## Usage
+
+Some example usage patterns of `Alert`.
+
+### Without title
+
+<img src="./alert-without-title.png" alt="" width="520">
+
+```jsx
+<Alert variant="notice">
+  <Alert.Description>
+    The payment was made 6 months ago, therefore you can’t issue refund to this merchant.
+  </Alert.Description>
+</Alert>
+```
+
+### Full bleed
+
+> **Note**
+>
+> Design update pending to remove all borders and border radii
+
+This full bleed layout works for all desktop, mobile and native when the `isBorderless` prop is passed
+
+<img src="./alert-mobile-full-bleed.png" alt="" width="520">
+
+```jsx
+<Alert title="International Payments Only" isBorderless>
+  <Alert.Description>
+    Use vendor payouts to quickly generate invoices.{' '}
+    <Link href="https://razorpay.com">Know More</Link>
+  </Alert.Description>
+</Alert>
+```
+
+## Open questions
+
+**Q1.** Should we use a different variant naming convention in code like `success`, `warn`, `info`, `error`? Currently this is same as design
+
+**A1.** TBD
+
+**Q2.** It's possible to pass any JSX as content but ideally we want to restrict it to few components (like `Link`, `string`, `List`). Doing these on code side would mean lot of checks. Similarly, restriction on `Alert.SecondaryAction` which should always be used with `Alert.PrimaryAction`. How do we do these - TS / runtime checks, etc.? One simpler alternative is to rely on design side restrictions for these but it makes the API more flexible should someone want to try things different from design.
```

Comment:
we can put the type of children as `children: Link | List | string`? and that might balance off flexibility+constraints?

For restricting the Primary and Secondary action thing - i think that might be the roadblock for compound component in this case. because the pattern itself proposes that things shall be composable. if we decide to move away from compound component then we can probably accept one prop similar to reshaped for actions and then add restrictions on that prop

Reply:
Defining types on `children` is a good start, though I'm not sure if it'll be able to cover all possible combinations. For example - there can be a `Link` inside `List` content, so we would need sort of nested checks, not to forget that these are TS only checks so people using JS wouldn't get any constraints on these checks 😅 

Having an `actions` prop does help with restriction on preventing users from having a secondary action without primary action but it comes with its own learning curve:
- Actions should be an array of max 2 objects (first element goes as primary action and second element goes as secondary action)
- Secondary action can also have `href` because it's a link button, so the second object in actions should have a key to pass that (so users need to remember the order first is always a primary button and second is always a link button)
- Users might think they can pass another object to render 3 or more actions in the row 😅 

I put these as compound components so it feels more natural (*in declarative JSX way*) to compose and pass props to them like `onClick`, `children`, `href`.

But compound components do have their own learning curve so I don't have strong opinions to choose one over another; though personally favoring them slightly more. Open to other opinions, we can decide and go forward either way.

Reply:
> not to forget that these are TS only checks so people using JS wouldn't get any constraints on these checks

In fact, you can't do TS checks on JSX childrens. Since children types get widened to JSX.Element so you won't be able to restrict it statically. This is a current design limitation of how JSX is integrated with TS. 

> Having an actions prop does help with restriction on preventing users from

Why accept array here? If we accept object most the the issues which you pointed will be solved. 

```tsx
<Alert actions={{ primary: {}, secondary: {} }} />
```
Or 

```tsx
<Alert primaryAction={} secondaryAction={} />
```

Reply:
Using array for actions would push people towards always specifying the primary action before using secondary action because primary action would be first element in the array.

Reply:
If we can't do TS checks to restrict passed content to link, list and string then runtime checks seem like the only option (which I feel might be an overkill right now 😅 )

Reply:
Arey yeah I got it. 
But with object we will be in control. 

Doesn't matter if users do `{ secondary: {}, primary: {} }` or `{ primary: {}, secondary: {} }`
We will render it our way inside the component.

Reply:
Ideally we want to restrict using secondary only when primary is also being used (not on its own)